### PR TITLE
feat: add password change interval to organization

### PR DESCRIPTION
**Pull request by dplynskiy**
- add password change interval to organization
- replace passwordChangeRequired to passwordChangeTime for user
- check passwordChangeTime at login
- Password change required field at user page fill  passwordChangeTime with now()
- when updating organization password change interval -> recalculate passwordChangeTime for all user in org except users with passwordChangeTime not null and  < now()

### CODE REVIEW

File: object/organization.go [hunk 1]
```diff
@@ -495,3 +506,32 @@ func (org *Organization) GetInitScore() (int, error) {
...
+		err := ormer.Engine.Table("user").Where("password_change_time >= now() or password_change_time is null and owner = ?", owner).Find(&users)
+		if err != nil {
+			return err
+		}
+		for _, user := range users {
+			if user.PasswordChangeTime.IsZero() {
+				user.PasswordChangeTime = getNextPasswordChangeTime(newInterval)
+			} else {
+				user.PasswordChangeTime = user.PasswordChangeTime.
+					Add(-1 * time.Hour * 24 * time.Duration(oldInterval)).
```
Comment (ISulimanov) [inline object/organization.go:526]: @dplynskiy can you move it to variable with clear name?

File: object/organization.go [hunk 2]
```diff
@@ -495,3 +506,32 @@ func (org *Organization) GetInitScore() (int, error) {
...
+					Add(time.Hour * 24 * time.Duration(newInterval))
+			}
+
+			_, err := ormer.Engine.ID(core.PK{user.Owner, user.Name}).Cols("password_change_time").Update(user)
```
Comment (ISulimanov) [inline object/organization.go:530]: can we use tx?

File: object/user.go [hunk 3]
```diff
@@ -224,8 +225,12 @@ type ManagedAccount struct {
 	SigninUrl   string `xorm:"varchar(200)" json:"signinUrl"`
 }
 
+func (u *User) PasswordChangeRequired() bool {
+	return u.PasswordChangeTime.Before(time.Now()) && !u.PasswordChangeTime.IsZero()
```
Comment (ISulimanov) [inline object/user.go:229]: swap

File: object/user.go [hunk 4]
```diff
@@ -727,6 +751,10 @@ func AddUser(user *User) (bool, error) {
 
 	user.PreHash = user.Hash
 
+	if organization.PasswordChangeInterval != 0 && user.PasswordChangeTime.IsZero() {
```
Comment (ISulimanov) [inline object/user.go:754]: swap

Comment (dplynskiy) [reply object/user.go]: done

File: object/organization.go (hunk 2 above)
Comment (dplynskiy) [reply object/organization.go]: yep, now updateOrganization method entirely works in transaction

File: object/organization.go (hunk 1 above)
Comment (dplynskiy) [reply object/organization.go]: moved to getIntervalFromdays in user_util

**Review by Aleksgav — APPROVED** [review]

**Review by ISulimanov — APPROVED** [review]

**Review by ISulimanov — APPROVED** [review]

**Pull request merged**